geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -750,10 +750,11 @@ protected int sizeValue(OrderedSetEntry value) {
 
     public void toData(DataOutput out) throws IOException {
       InternalDataSerializer.writePrimitiveInt(size(), out);
-      for (int pos = n; pos-- != 0;) {
-        if (value[pos] != null) {
-          byte[] member = value[pos].getMember();
-          byte[] score = value[pos].getScoreBytes();
+      for (int pos = getMaxIndex(); pos-- != 0;) {
```

Comment:
Since there needs to be one more round of changes to fix the integration tests, could this for loop instead be:
```
for (int pos = getMaxIndex() - 1; pos >= 0; pos--) {
```
to make it more consistent with the rest of the module? It's a little clunkier, but for me, at least, is easier to understand at a glance. Alternately, since the order in which we iterate through the collection doesn't matter here, it could be:
```
for (int i = 0; i < getMaxIndex(); ++i) {
```
which is much clearer, I think.
The same would apply to the loop in `RedisHash.toData()` if you choose to change this

Reply:
The for loop expression came from the original code. I just added the accessor method getMaxIndex instead of using "n". I agree your way is easier to understand. I was worried about changing it and introducing an off by 1 error.